README.md

```diff
@@ -349,6 +349,33 @@ In [the definition for HTML](https://github.com/whatwg/html/pull/4571), the call
 
 The WeakRefs proposal guarantees that multiple calls to `WeakRef.prototype.deref()` return the same result within a certain timespan: either all should return `undefined`, or all should return the object. In HTML, this timespan runs until a [microtask checkpoint](https://html.spec.whatwg.org/multipage/webappapis.html#perform-a-microtask-checkpoint), where HTML performs a microtask checkpoint when the JavaScript execution stack becomes empty, after all Promise reactions have run.
 
+In other words, for a WeakRef `wr`, synchronous code like the following is guaranteed to return the same result for `wr.deref()`:
+
+```js
+if (wr.deref() !== undefined) {
+  // […]
+  wr.deref().foo();
+  // […]
+  wr.deref().bar();
+  // […]
+}
+```
+
+The guarantee goes away if any of the blanks (`[…]`) introduces a microtask checkpoint, for example if it contains an `await` expression, or if it wraps the remainder of the code in a promise callback (e.g. `Promise.resolve().then(() => […])`).
```

Comment:
> microtask checkpoint

This is not an EcmaScript concept.

> or if it wraps the remainder of the code in a promise callback (e.g. `Promise.resolve().then(() => […])`)

probably worth saying somehow. But literally this doesn't make syntactic sense for the example code, as it is shown as straight-line code.

Reply:
This is re-using the same "microtask checkpoint" terminology from the previous paragraph. It's indeed HTML spec jargon. I'm definitely open to rephrasing it to make it more general; could you recommend a better phrasing?

> But literally this doesn't make syntactic sense for the example code, as it is shown as straight-line code.

It only makes sense if you think of each `// […]` as an injection point where arbitrary text can go. One could then replace the first `// […]` with:

```
Promise.resolve().then(() => {
```

...and the last `// […]` with:

```js
});
```

Any idea on how to make this more clear? Or should we just drop this particular example?

Reply:
> This is re-using the same "microtask checkpoint" terminology from the previous paragraph. It's indeed HTML spec jargon. I'm definitely open to rephrasing it to make it more general; could you recommend a better phrasing?

Thanks for pointing out the previous paragraph. That language needs to go too. EcmaScript has no tasks or microtasks. As of the last tc39 mtg, @littledan and I arrived at a mutual understanding about how to phrase all of this in terms of Job scheduling and Job execution. I'll first let @littledan speak for his memory of our mutual understanding and how it bears on this issue before making my own suggestions.

@littledan , did we record our mutual understanding? Was it in your subsequent talk slides? Meeting notes? Thanks.

Reply:
> Any idea on how to make this more clear? Or should we just drop this particular example?

I see what you're doing and it is coherent and informative. However, you're filling in the `[...]`s with unbalanced bracketing, which would change the indentation level. That's what's confusing. Although not an official part of the syntax, we all understand code in terms of indentation level.

Yes, I think the best thing is probably just to drop this particular example, assuming everyone understands the shown indentation level to rule it out.

Reply:
> Thanks for pointing out the previous paragraph. That language needs to go too.

Did you see the previous paragraph? It explicitly points out that this is an HTML concept, which seems fine by me.

Either way, I'd love to hear more about the discussion you had with Dan. He's on vacation now so I imagine it'll take a while until we hear back.

> I see what you're doing and it is coherent and informative. However, you're filling in the [...]s with unbalanced bracketing, which would change the indentation level. That's what's confusing. Although not an official part of the syntax, we all understand code in terms of indentation level.
>
> Yes, I think the best thing is probably just to drop this particular example, assuming everyone understands the shown indentation level to rule it out.

Ack. Removed this example.